exporter/datadogexporter/translate_traces.go

```diff
@@ -93,7 +93,14 @@ func convertToDatadogTd(td pdata.Traces, fallbackHost string, cfg *config.Config
 			host = fallbackHost
 		}
 
-		seenHosts[host] = struct{}{}
+		if host != "" {
+			seenHosts[host] = struct{}{}
+		} else {
+			tags := attributes.RunningTagsFromAttributes(rs.Resource().Attributes())
```

Comment:
Can you please double-check this? What's the relation between the host being empty and tags being marked as "seen" based on resource attributes?

Reply:
Thanks for checking! It's correct, it's just that the naming is a bit confusing, sorry (will fix that in a follow up PR since it needs changes elsewhere). 

For a given payload we want to track its 'source' (after this PR that is either a specific host or an ECS task). We never want to identify a single payload with two different sources: we only mark a payload coming from an ECS task if it doesn't have a hostname.